slides/11_State_Management.html

```diff
@@ -78,30 +79,33 @@ <h3>Step by Step</h3>
           </section>
 
           <section>
-            <pre>
-                    <code class="hljs tsx" data-trim>
-                        <script type="text/template">
+            <pre><code class="hljs tsx" data-trim><script type="text/template">
             import { createContext, useContext } from "react";
 
-            type YourContextValue = {
-              user: string;
+            type ContextType = {
+              name: string;
+              setName: (name: string) => void;            
             };
             
-            const UserContext = createContext<YourContextValue | undefined>(undefined);
+            const UserContext = createContext<ContextType | undefined>(undefined);
             
             function Parent() {
-              return (<UserContext.Provider value={{ user: "joe" }}>
-                <Child />
-              </UserContext.Provider>)
+              const [name, setName] = useState("Joe");
+              return (
+                <UserContext.Provider value={{ name, setName }}>
+                  <Child />
+                </UserContext.Provider>)
             }
 
             function Child() {
               const context = useContext(UserContext);
-              return <p>{context?.user}</p>; // joe
+              const updateName = () => context?.setName("Jane")
```

Comment:
I would usually do such a small onClick handle inline. But if we do it outside, in my opinion we should stick to the explicit function definition instead of anonymous const function definition. Explicit function definition are encouraged/suggested in the beta React docs and can be better debugged as their name appears in the stack trace.

I.e. `function updateName() { ... }`